### DISCUSSION TITLE
Is it possible to support Hybrid AI cards ?

### DISCUSSION THREAD

**Discussion by lileidev** [ROCm/ROCm#2605] [Ideas]
If there are more than one type of AI cards on a server, is it possible to comminucate between these cards?
Or if have many AI cards clusters, such as cluster A with AI card type a, cluster B with AI card type b, cluster C with AI card type c, is it possible to combine a super cluster for training models?

**Comment by fluidnumericsJoe** [comment]
What is an AI card?

**Reply by lileidev** [reply]
Such as NVIDIA A100， H100, AMD MI300, Intel Habana Gaudi...   It means an accelerate card for AI model training or inference.

**Comment by fluidnumericsJoe** [comment]
So long as your application can pick at runtime the correct compiled kernels for the target GPU, this is possible. If your application is tightly coupled and communication is required between GPUs, you will very likely run into problems. I wouldn't call such a cluster a "super cluster"; it's more "scrappy" (which I am fond of) and requires some careful thought when figuring out how to deploy your applications on such a heterogeneous system
